src/main/java/duke/system/Parser.java

```diff
@@ -0,0 +1,172 @@
+    private static final String LINE = "\n____________________________________________________________";
+
+    enum predefinedCommand {
+        LIST,
+        BYE,
+        DONE,
+        TODO,
+        DEADLINE,
+        EVENT,
+        ERROR,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * Constructor method that initiates with empty/ null values
+     */
+    public Parser() {
+        this.command = "";
+        this.date = null;
+        this.argument = "";
+    }
+
+    /**
+     *
+     * @param in - a string that will be parsed and stored as <code>command</code>, <code>argument</code> and <code>date</code> accordingly
+     * @throws DukeException.UnknownCommandException if unknown command entered
+     * @throws DukeException.NoDescriptionException if required no. of arg is not met
+     */
+    public Parser(String in) {
+        String tempDate = null;
+        String tempCommand = "";
+        String[] result = in.split("\\s");
+        String tempArg = "";
+        try {
+            if (result[0].equals("done") || result[0].equals("find")) {
+                tempCommand = result[0];
+                if (result.length <= 1) {
+                    throw new DukeException.NoDescriptionException(result[0]);
+                } else {
+                    tempArg = result[1];
+                }
+            } else if (result[0].equals("todo") || result[0].equals("delete")) {
+                String temp = in.substring(in.indexOf(" ") + 1);
+                tempArg = temp;
+                if (temp.equals("todo") || temp.equals("delete")) {
+                    throw new DukeException.NoDescriptionException(result[0]);
+                } else {
+                    tempCommand = result[0];
+                }
+            } else if (result[0].equals("deadline") || result[0].equals("event")) {
+                String firstParam = in.substring(in.indexOf("/") + 1);
+                if (firstParam.equals("deadline") || firstParam.equals("event")) {
+                    throw new DukeException.NoDescriptionException(result[0]);
+                } else {
+                    int dateIndex = Math.max(firstParam.indexOf("by "), firstParam.indexOf("at "));
+                    if (dateIndex == -1) {
+                        throw new DukeException.NoDescriptionException(result[0]);
+                    } else {
+                        tempCommand = result[0];
+                        tempDate = firstParam.substring(dateIndex + 3);
+                        firstParam = in.substring(in.indexOf(" ") + 1);
+                        tempArg = firstParam.substring(0, firstParam.indexOf("/") - 1);
+                    }
+                }
+            } else {
+                try {
+                    tempCommand = String.valueOf(predefinedCommand.valueOf(result[0]));
+                } catch (IllegalArgumentException ex) {
+                    throw new DukeException.UnknownCommandException();
+                }
+            }
```

Comment:
From the textbook -> Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.
I would suggest trying not to put all the logic in here, perhaps can create new methods in this class to handle certain common things! :)